### PR TITLE
EC-218: Checkout revision

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/EC-218
| Versions      | 3.1, 3.2, master

### CODE REVIEW

File: docs/guide/checkout/order_confirmation.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+In case of a failure (e.g. if the mail server is unreachable) the issue is logged into `var/log/<env>/dev-siso.mails.log`.
+
+Processing the confirmation email address is handled in `Siso\Bundle\CheckoutBundle\Service\SummaryCheckoutFormService`.
+
+### Customer confirmation
+
+The customer's confirmation email address depends on the user type and is stored in the basket object during the checkout process.
+
+For an anonymous customer the email address is taken from the invoice address of the basket object.
+For a logged-in customer the address is taken from customer profile data.
```
Comment (dabrt) [inline docs/guide/checkout/order_confirmation.md:24]:
```suggestion
For a logged-in customer, the address is taken from customer profile data.
```

File: docs/guide/checkout/order_confirmation.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+### Sales contact confirmation
+
+A second confirmation email can be sent to a sales contact.
+This address is also stored in the basket object next to the customer's address. The behavior is configurable:
```
Comment (dabrt) [inline docs/guide/checkout/order_confirmation.md:29]:
```suggestion
This address is also stored in the basket object, next to the customer's address. 
The behavior is configurable:
```

File: docs/guide/checkout/order_confirmation.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+### Shop owner
+
+Owner of the shop also receives a receives a confirmation email for a [local order](local_orders.md)
+if `is_shop_owner` is set to `true`.
+
+This email contains a special message and subject.
+
+- `shop_owner_mail_subject` (set in the email configuration)
+- `email_shop_owner_intro_text` (textmodule inside the backend)
```
Comment (dabrt) [inline docs/guide/checkout/order_confirmation.md:62]:
```suggestion
- `email_shop_owner_intro_text` (text module inside the backend)
```
Also - what does it mean "inside the backend"?

File: docs/guide/checkout/return_process.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Return process
+
+[[= product_name_com =]] offers a simple RMA (return merchandise authorization) process.
+The goal is to inform the user about the cancellation policies and give them the possible to return their goods, including online returns.
```
Comment (dabrt) [inline docs/guide/checkout/return_process.md:4]:
```suggestion
The goal is to inform the user about the cancellation policies and give them the possibility to return their goods, including online returns.
```

**Review by dabrt — CHANGES_REQUESTED** [review]

File: docs/guide/checkout/local_orders.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+parameters:
+    siso_local_order_management.default.send_order_to_erp: false
+```
+
+To avoid sending the order to ERP, the `LocalOrderRequiredException` exception is thrown that interrupts the default process (sending to ERP).
+The message and the order are logged as failed.
+
+The shop then reacts like an ERP system, confirms the order and returns an order ID.
+
+The response has the same structure as a response returned from ERP, so no additional changes in the template are required.
```
Comment (DominikaK) [inline docs/guide/checkout/local_orders.md:25]:
```suggestion
The response has the same structure as the response returned from ERP, so no additional changes in the template are required.
```
As this is a specific response (referred to by "returns an order ID" in the previous line)

File: docs/guide/checkout/order_confirmation.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+After finishing the checkout (in case of an electronic payment transaction after redirection from the payment provider) the user is redirected to the confirmation page.
+
+If the customer has a customer number, customer data is fetched from the ERP again.
+
+## Confirmation email
+
+The `Siso\Bundle\CheckoutBundle\EventListener\OrderConfirmationListener` event listener sends order confirmation
+as soon as the ERP accepts the order.
+
+Both the customer and a sales contact (if configured) receive confirmation emails.
```
Comment (DominikaK) [inline docs/guide/checkout/order_confirmation.md:12]: How about:
```suggestion
Both the customer and a sales contact (if configured) should receive confirmation emails.
```

File: docs/guide/checkout/order_confirmation.md (hunk 3 above)
Comment (DominikaK) [inline docs/guide/checkout/order_confirmation.md:62]:
```suggestion
 `email_shop_owner_intro_text` (text module in the Back Office)
```

File: docs/guide/checkout/order_confirmation.md (hunk 6 above)
Comment (dabrt) [reply docs/guide/checkout/order_confirmation.md]: ok

**Review by dabrt — APPROVED** [review]

**Pull request merged**